### PR TITLE
Added macro to create functions in PipelineNamespace

### LINKED ISSUE #26380: Write a macro to generate namespaced Ids
We have a bunch of Indexes and Ids in `components/msg/constellation_msg.rs`, see for example the `BrowsingContextIndex` and `BrowsingContextId` https://github.com/servo/servo/blob/1b4162aecbb935c8183438c3d6bc285db3932476/components/msg/constellation_msg.rs#L263

I find myself manually adding new ones regurlarly by hand, perhaps someone wants to write a macro for this? See for example the changes to this file: https://github.com/servo/servo/pull/26317/files#diff-8dc9268416cd4f037d50206e76c73cf9

I'd be great if we could just do something along the lines of `namespaced_id!(ServiceWorkerId)`.

Note that each also require implementing a method on `PipelineNamespace`, which I guess will have to be done by hand(unless someone has a better idea). 

I don't have much, if any, experience writing macros, but I'll venture ahead with a "less easy" label...

### PR DESCRIPTION
**Pull request by AbhishekSharma102**
Changes to constellation_msg.rs to generate namespace id using macros

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #26380 
- [X] These changes do not require tests because no functionality is changed.

### CODE REVIEW

**Review by gterzian — CHANGES_REQUESTED** [review]
Thank you, looks likes a great start!

Would it be possible to additional functionality to the macro, maybe via a separate one, that would generate the various `Id` types, something like `namespaced_id!(PipelineId)`(which would then also generate the matching `PipelineIndex`)?

Perhaps this could be done with the same macro, but I am not sure, in that case it could be for example `create_pipeline_namespace! {PipelineId, self}` that would do what the current macro does, plus would also generate the `PipelineId` and `PipelineIndex` types.

**Comment by AbhishekSharma102** [comment]
Hey!

 I believe rust does not allow modification to variable names (like suffix and prefix). I was wondering if we could pass `Pipeline` to the macro, and it would append `Id` to get `PipelineId` and `Index`  to get `PipelineIndex` . But rust does not allow it. So we would need to pass all the variable names. 

We can definitely generate the Id and Index inside the macros. I am thinking of having two macros, one defined in the PR, and another one `namespace_id` which takes two args, first is `id_name` like `PipelineId` and second `index_name` like `PipelineIndex` and returns them.

Let me know if this sounds good.

**Comment by gterzian** [comment]
> So we would need to pass all the variable names.

Ok, so you mean we'd have to pass `PipelineId` and `PipelineIndex`, sounds good to me, definitely better than the current situation.

> I am thinking of having two macros, one defined in the PR, and another one namespace_id which takes two args, first is id_name like PipelineId and second index_name like PipelineIndex and returns them.

Sounds good. 

> Also I believe two macros would need to be called separately. Please correct if I'm wrong.

I'm not sure, let me know what you find out!

**Comment by AbhishekSharma102** [comment]
Hey, I have added the macro. Let me know if this looks good.

File: components/msg/constellation_msg.rs [hunk 1]
```diff
@@ -136,68 +163,17 @@ impl PipelineNamespace {
...
-        }
-    }
-
-    fn next_blob_id(&mut self) -> BlobId {
-        BlobId {
-            namespace_id: self.id,
-            index: BlobIndex(self.next_index()),
-        }
-    }
+    create_pipeline_namespace_function! {next_pipeline_id, PipelineId, self, PipelineIndex}
```
Comment (gterzian) [inline components/msg/constellation_msg.rs:166]: nit: how about we name it a bit shorter, like `namespace_id_method`?

**Review by gterzian — CHANGES_REQUESTED** [review]
Looks great! Just one question.

File: components/msg/constellation_msg.rs (hunk 1 above)
Comment (AbhishekSharma102) [reply components/msg/constellation_msg.rs]: renamed it

**Comment by gterzian** [comment]
Great! If you can please squash the third commit into the first, then it should be good to go!

**Comment by AbhishekSharma102** [comment]
Squashed them into one.

**Comment by gterzian** [comment]
Awesome, thanks!

@bors-servo r+

**Comment by AbhishekSharma102** [comment]
@gterzian Can you please suggest another issue that I can take up?

**Comment by gterzian** [comment]
I'm not sure, there are not many issues marked "less easy" any more, perhaps this one: https://github.com/servo/servo/issues/24684?

If you want something harder I think this one is interesting: https://github.com/servo/servo/issues/26571 (it might require some more guidance)

**Comment by AbhishekSharma102** [comment]
I'd like to work on #26571 as I'm hoping there will be more learning. Could you please assign that to me? And please tell on what to read up for the same?

**Pull request merged**